zanata-war/src/main/java/org/zanata/ApplicationConfiguration.java

```diff
@@ -408,4 +415,33 @@ public String copyrightNotice() {
         return msgs.format("jsf.CopyrightNotice",
                 String.valueOf(Calendar.getInstance().get(Calendar.YEAR)));
     }
+
+    @Observer(PostAuthenticateEvent.EVENT_NAME)
+    public void setAuthenticatedSessionTimeout(
+            @Observes PostAuthenticateEvent payload) {
+        HttpServletRequest request = ServletContexts
+                .getInstance()
+                .getRequest();
+        if (request != null) {
+            request
+                    .getSession()
+                    .setMaxInactiveInterval(
+                            max(authenticatedSessionTimeoutMinutes * 60,
+                                    defaultAnonymousSessionTimeoutMinutes *
```

Comment:
I think this would be a little easier to read if you defined a local variable (eg timeout) and calculated it a few lines up.  (Or if the setter methods adjusted authenticatedSessionTimeoutMinutes when crazy values come in.)